cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge_v2.rs

```diff
@@ -315,156 +264,215 @@ fn transfer_relay_token() {
-			vec![RuntimeEvent::XcmpQueue(cumulus_pallet_xcmp_queue::Event::XcmpMessageSent { .. }) =>{},]
+			vec![RuntimeEvent::EthereumOutboundQueueV2(snowbridge_pallet_outbound_queue_v2::Event::MessageQueued{ .. }) => {},]
 		);
 	});
+}
 
-	AssetHubWestend::execute_with(|| {
-		type RuntimeEvent = <AssetHubWestend as Chain>::RuntimeEvent;
-		type RuntimeOrigin = <AssetHubWestend as Chain>::RuntimeOrigin;
+#[test]
+fn transact_with_agent() {
+	let weth_asset_location: Location =
+		(Parent, Parent, EthereumNetwork::get(), AccountKey20 { network: None, key: WETH }).into();
 
-		// Check that AssetHub has issued the foreign asset
-		assert_expected_events!(
-			AssetHubWestend,
-			vec![RuntimeEvent::ForeignAssets(pallet_assets::Event::Issued { .. }) => {},]
-		);
+	fund_sovereign();
 
-		// Local fee amount(in DOT) should cover
-		// 1. execution cost on AH
-		// 2. delivery cost to BH
-		// 3. execution cost on BH
-		let local_fee_amount = 200_000_000_000;
-		// Remote fee amount(in WETH) should cover execution cost on Ethereum
-		let remote_fee_amount = 4_000_000_000;
+	register_weth();
+
+	BridgeHubWestend::execute_with(|| {});
+
+	AssetHubWestend::execute_with(|| {
+		type RuntimeOrigin = <AssetHubWestend as Chain>::RuntimeOrigin;
 
 		let local_fee_asset =
-			Asset { id: AssetId(Location::parent()), fun: Fungible(local_fee_amount) };
-		let remote_fee_asset =
-			Asset { id: AssetId(weth_asset_location.clone()), fun: Fungible(remote_fee_amount) };
+			Asset { id: AssetId(Location::parent()), fun: Fungible(LOCAL_FEE_AMOUNT_IN_DOT) };
+
+		let remote_fee_asset = Asset {
+			id: AssetId(weth_asset_location.clone()),
+			fun: Fungible(REMOTE_FEE_AMOUNT_IN_WETH),
+		};
 		let reserve_asset = Asset {
 			id: AssetId(weth_asset_location.clone()),
-			fun: Fungible(TOKEN_AMOUNT - remote_fee_amount),
+			fun: Fungible(TOKEN_AMOUNT - REMOTE_FEE_AMOUNT_IN_WETH),
 		};
 
-		let weth_asset =
-			Asset { id: weth_asset_location.clone().into(), fun: Fungible(TOKEN_AMOUNT) };
-		let dot_asset = Asset { id: AssetId(Location::parent()), fun: Fungible(TOKEN_AMOUNT) };
-
-		let assets = vec![weth_asset, dot_asset.clone(), local_fee_asset.clone()];
-		let destination = Location::new(2, [GlobalConsensus(Ethereum { chain_id: CHAIN_ID })]);
+		let assets = vec![
+			Asset { id: weth_asset_location.clone().into(), fun: Fungible(TOKEN_AMOUNT) },
+			local_fee_asset.clone(),
+		];
 
-		let beneficiary = Location::new(
-			0,
-			[AccountKey20 { network: None, key: ETHEREUM_DESTINATION_ADDRESS.into() }],
-		);
+		let beneficiary =
+			Location::new(0, [AccountKey20 { network: None, key: AGENT_ADDRESS.into() }]);
 
-		let xcm_on_bh = Xcm(vec![DepositAsset { assets: Wild(All), beneficiary }]);
+		let call_params = CallContractParams {
+			target: Default::default(),
+			data: vec![],
+			gas_limit: 40000,
+			// value should be less than the transfer amount, require validation on BH Exporter
+			value: 4 * (TOKEN_AMOUNT - REMOTE_FEE_AMOUNT_IN_WETH) / 5,
+		};
```

Comment:
Require fund agent in same message, The Ether `value` should be less than amount of WETH to prefund.